pkg/controller/operators/catalog/operator.go

```diff
@@ -1168,6 +1169,9 @@ func (o *Operator) ensureInstallPlan(logger *logrus.Entry, namespace string, gen
 		return nil, err
 	}
 
+	o.muInstallPlan.Lock()
+	defer o.muInstallPlan.Unlock()
```

Comment:
@njhale Any idea on whether this might also solve the duplicate installplan resources getting generated when a subscription gets created? I can't seem to dig up the BZ anymore (maybe we closed that out as wontfix at some point?).